### PR TITLE
tools/scanner: Add support for SPDX scanning.

### PR DESCRIPTION
**Pull request by oliverchang**
Also refactor the existing scanner to conform to standard Go project
structure.

First part of #257

### CODE REVIEW

**Comment by oliverchang** [comment]
@calebbrown mind reviewing this Go PR for OSV?

File: tools/scanner/cmd/scanner.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+			log.Printf("Scanned %s SBOM", provider.Name())
+			return nil
+		}
+
+		if errors.Is(err, sbom.InvalidFormat) {
+			continue
+		}
+
+		return err
+	}
```
Comment (calebbrown) [inline tools/scanner/cmd/scanner.go:63]: Moving this logic into its own function make make the code easier to read (e.g. `scanFile()`)

File: tools/scanner/cmd/scanner.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	cmd.Stdout = &out
+
+	err := cmd.Run()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(out.String()), nil
+}
+
+func doScan(repoDir string) error {
```
Comment (calebbrown) [inline tools/scanner/cmd/scanner.go:99]: How does `doScan()` differ to `scan()`? The names imply they are the same - is there a better name for this one?

**Review by calebbrown — APPROVED** [review]

File: tools/scanner/cmd/scanner.go (hunk 1 above)
Comment (oliverchang) [reply tools/scanner/cmd/scanner.go]: Done!

File: tools/scanner/cmd/scanner.go (hunk 2 above)
Comment (oliverchang) [reply tools/scanner/cmd/scanner.go]: Good point. renamed `doScan` to `scanGit`.

**Pull request merged**